dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorConfiguration.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents function choice behavior configuration produced by a <see cref="FunctionChoiceBehavior" />.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class FunctionChoiceBehaviorConfiguration
+{
+    /// <summary>
+    /// Creates a new instance of the <see cref="FunctionChoiceBehaviorConfiguration"/> class.
+    /// </summary>
+    internal FunctionChoiceBehaviorConfiguration()
+    {
+    }
+
+    /// <summary>
+    /// Represents an AI model's decision-making strategy for calling functions.
+    /// </summary>
+    public FunctionChoice Choice { get; internal set; }
+
+    /// <summary>
+    /// The functions available for AI model.
+    /// </summary>
+    public IEnumerable<KernelFunction>? Functions { get; internal set; }
```

Comment:
We may want to expose `Count` property instead of enforcing iterations over the functions.

```suggestion
    public ICollection<KernelFunction>? Functions { get; internal set; }
```